src/view/components/decisionScopesComponent.jsx

```diff
@@ -10,116 +10,99 @@ OF ANY KIND, either express or implied. See the License for the specific languag
+            Provide data element returning array of scopes
+          </Radio>
+        </RadioGroup>
       </div>
-      {values.decisionsInputMethod === options.DATA_ELEMENT && (
-        <div className="FieldSubset u-gapTop">
-          <div>
-            <WrappedField
+      {decisionsInputMethod === DATA_ELEMENT && (
+        <div className="FieldSubset">
+          <DataElementSelector name="decisionScopesDataElement">
+            <TextField
               data-test-id="scopeDataElementField"
-              id="scopeDataElementField"
               name="decisionScopesDataElement"
-              component={Textfield}
-              componentClassName="u-fieldLong"
-              supportDataElement="replace"
+              width="size-5000"
             />
-          </div>
+          </DataElementSelector>
         </div>
       )}
-      {values.decisionsInputMethod === options.CONSTANT && (
-        <div className="FieldSubset u-gapTop">
-          <FieldArray
-            name="decisionScopesArray"
-            render={arrayHelpers => {
-              return (
-                <div className="u-gapTop">
-                  {values.decisionScopesArray.map((scope, index) => (
-                    <div className="u-gapTop" key={index}>
-                      <WrappedField
-                        data-test-id={`scope${index}Field`}
-                        id={`scope${index}Field`}
-                        name={`decisionScopesArray.${index}`}
-                        component={Textfield}
-                        componentClassName="u-fieldLong"
-                      />
-                      <Button
-                        data-test-id={`deleteScope${index}Button`}
-                        icon={<Delete />}
-                        disabled={values.decisionScopesArray.length === 1}
-                        variant="tool"
-                        onClick={() => {
-                          arrayHelpers.remove(index);
-                        }}
-                      />
-                    </div>
-                  ))}
-                  <div className="u-gapTop">
-                    <Button
-                      data-test-id="addDecisionScopeButton"
-                      label="Add scope"
-                      onClick={() => {
-                        arrayHelpers.push("");
-                      }}
-                    />
-                  </div>
-                </div>
-              );
-            }}
-          />
+      {decisionsInputMethod === CONSTANT && (
+        <div className="FieldSubset">
+          {decisionScopeFields.map((field, index) => {
+            return (
+              <div className="u-gapBottom" key={field.id}>
+                <TextField
+                  data-test-id={`scope${index}Field`}
+                  name={`decisionScopesArray.${index}.value`}
+                  width="size-5000"
+                  defaultValue={field.value}
```

Comment:
This is kind of another gotcha with React Hook Form. When dealing with field arrays, you have to set `defaultValue` because of how it optimizes rendering of items. https://github.com/react-hook-form/react-hook-form/discussions/5048